pinecone/client.go

```diff
@@ -107,37 +107,24 @@ func (c *Client) IndexWithAdditionalMetadata(host string, namespace string, addi
 	return idx, nil
 }
 
-func (c *Client) extractAuthHeader() map[string]string {
-	possibleAuthKeys := []string{
-		"api-key",
-		"authorization",
-		"access_token",
-	}
-
-	for key, value := range c.headers {
-		for _, checkKey := range possibleAuthKeys {
-			if strings.ToLower(key) == checkKey {
-				return map[string]string{key: value}
-			}
-		}
-	}
-
-	return nil
-}
-
 func (c *Client) ListIndexes(ctx context.Context) ([]*Index, error) {
 	res, err := c.restClient.ListIndexes(ctx)
 	if err != nil {
 		return nil, err
 	}
 	defer res.Body.Close()
 
+	resBodyBytes, err := io.ReadAll(res.Body)
```

Comment:
Ran into EOF problems when trying to read the body in multiple places. To avoid that I went with reading the body initially and then passing `[]byte` around. Let me know if this is inefficient or confusing in some way, maybe there's a better way to be doing this.